### ISSUE TITLE
[VM Agent Templates] Switch Windows Server 2019 Container to Windows Server 2019 Core and customize Container Runtime install

### ISSUE THREAD

**Issue by dduportal** [jenkins-infra/helpdesk#2942]
### Service(s)

Other

### Summary

As notified by @timja , and described by Microsoft in https://techcommunity.microsoft.com/t5/containers/updates-to-the-windows-container-runtime-support/ba-p/2788799, the Azure VM image providing the Docker Engine with Windows container out of the box won't be updated, and are scheduled for deletion in September 2023 from the catalog.

We have to update our [packer template for Windows](https://github.com/jenkins-infra/packer-images/blob/main/build-jenkins-agent-windows.pkr.hcl#L17) to stop using this serie of images, and to use the Windows 2019 Core Server line, which implies installing the Docker CE Engine for Windows container as part of the provisioning, to keep the actual behavior.

**Comment by dduportal** [comment]
Hopefully, there is some work on that area in https://github.com/jenkins-infra/packer-images/pull/219.

Tl;DR;

```powershell
Install-PackageProvider -Name NuGet -Force
Install-Module -Name DockerMsftProvider -Repository PSGallery -Force
Install-Package -Name docker -ProviderName DockerMsftProvider
## A reboot is required to use start containers: only check the client
docker -v ## client version only
```

(from https://docs.microsoft.com/en-us/virtualization/windowscontainers/quick-start/set-up-environment?tabs=Windows-Server#install-docker)


Might be interested to check if chocolatey doesn't support it (since it's using nuget), that would be even easier thanks to @lemeurherve 's work.

**Comment by dduportal** [comment]
Closing as it was deployed successfully:

- https://ci.jenkins.io/job/Infra/job/acceptance-tests/job/check-agent-availability/966/console
- https://github.com/jenkins-infra/jenkins-infra/pull/2184
